### DISCUSSION TITLE
Android ≤ 7 BRouter CertPathValidatorException

### DISCUSSION THREAD

**Discussion by Cinemannix** [devemux86/cruiser#457] [Q&A] [answered]
Hi there, on the phone from my girlfreind, I got error when we try to download new segemnts.. Error is java.security.cert.certpathvalidatorexception trust anchor for certification path.

Any ideas ?

Cine

**Comment by devemux86** [comment]
@Cinemannix thanks for the report.

What Android and device?

What is selected in "Settings | Routing | Options | Advanced"?
Internal or External BRouter?

**Comment by Cinemannix** [comment]
Samsung S5 NEO
internal

**Comment by Cinemannix** [comment]
Android 6.0.1

**Comment by Cinemannix** [comment]
to old ?

**Reply by devemux86** [reply]
Not too old, as I test and can run all functions even on Android 5.

Can you try with a clean installation of Cruiser?
Does the phone have a proxy or firewall or any network change?

**Comment by Cinemannix** [comment]
ok, i`ll try so with new. There is only wifi without no proxy

**Reply by devemux86** [reply]
> There is only wifi without no proxy

I mean in the phone settings if there is any special change in the network settings.

**Reply by Cinemannix** [reply]
no, no specific

**Reply by Cinemannix** [reply]
same problem after reinstall

**Reply by Cinemannix** [reply]
maps download ok
but no segments

**Comment by devemux86** [comment]
Can you download offline maps?

Does this happen with the **internal** BRouter downloader inside Cruiser (see image)?

Can you also try with the external BRouter application?

<img src="https://github.com/devemux86/cruiser/assets/3484020/cc264cf1-6c43-4576-a356-ee5dc4a066b5" width="50%"/>

**Comment by devemux86** [comment]
As an alternative, when you download Mapsforge maps, the BRouter data for these maps are also downloaded automatically.
This should work?

It just does not solve why the BRouter downloader does not work on that device.

Also what happens with the external BRouter app, does its downloader work?

**Reply by Cinemannix** [reply]
f*ck...the same...

**Reply by devemux86** [reply]
With the external BRouter app?

**Reply by Cinemannix** [reply]
jepp

**Reply by Cinemannix** [reply]
on my phone it works

**Reply by devemux86** [reply]
@Cinemannix do you mean that on Android 6 phone,
when download offline maps, only the maps are downloaded,
but not the BRouter data?

Is there a similar message? How do you know it?

**Reply by Cinemannix** [reply]
Hi, all the maps are no Problem to downlaod. Only when I try to download segments for Brouter. Try to make fotos

**Reply by devemux86** [reply]
@Cinemannix 

When **Mapsforge** maps are downloaded, then also the BRouter data for these maps are downloaded.
This happens automatically and with a different way than the BRouter downloader with the grid.

That's why I ask if this alternative way works? Does the routing works after completing the map download?
If it works then you can have offline routing on that device.

**Reply by Cinemannix** [reply]
<img src="https://github.com/devemux86/cruiser/assets/143206729/d22b3689-3243-4965-9942-d0aac99db02b" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/143206729/164b134b-9e7c-441b-bd23-7964d6c15f9c" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/143206729/a6d3e29a-181a-4ae3-b1e3-e9f6ae72bacc" width="33%"/>

**Reply by Cinemannix** [reply]
We will see, when we first time go out for a tour.

**Reply by devemux86** [reply]
@Cinemannix thanks for the screenshots! :+1:

**Comment by devemux86** [comment]
Let's me try some tests...

For now, does the alternative way works? See here:
- https://github.com/devemux86/cruiser/discussions/457#discussioncomment-8583831

**Comment by devemux86** [comment]
The good news is that I managed to reproduce it on Android 6,
so it will be easier to check what happens. :slightly_smiling_face:

**Comment by devemux86** [comment]
After more examination, I can reproduce it on Android 6 and 7.
(it seems to work on Android 5 and 8)

I reported it in BRouter repository. BRouter server manager was also notified.
- https://github.com/abrensch/brouter/issues/676

---

@Cinemannix 

I can do a workaround inside Cruiser so the **internal** BRouter can work again on old Android.
It will be available in the next update.

**Reply by Cinemannix** [reply]
Hi, thx for your work, but it sounds like a problem of Brouter. So they have to fix it. Cool that you want to fix it.

**Reply by devemux86** [reply]
I can fix it in the **internal** BRouter inside Cruiser, that most users should use by default,
because I can update it more often, has faster routing and can calculate longer routes.

The external BRouter app is separate, so they need to see what can be done.

**Comment by devemux86** [comment]
@Cinemannix it should work in the latest:
- https://github.com/devemux86/cruiser/discussions/451

**Reply by Cinemannix** [reply]
Ok, I can Check not until thursday

**Comment by devemux86** [comment, marked as answer]
BRouter downloads on old Android versions are fixed in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
